pkg/cosign/verify.go

```diff
@@ -432,6 +432,16 @@ func ValidateAndUnpackCertWithChain(cert *x509.Certificate, chain []*x509.Certif
 	return ValidateAndUnpackCert(cert, co)
 }
 
+// ValidateAndUnpackCertWithCertPools creates a Verifier from a certificate. Verifies that the certificate
+// chains up to the provided root. CheckOpts should contain a pool of CA Roots and optionally the Intermediates.
+// Optionally verifies the subject and issuer of the certificate.
+func ValidateAndUnpackCertWithCertPools(cert *x509.Certificate, co *CheckOpts) (signature.Verifier, error) {
```

Comment:
Given this is only used in one place, can we just call `ValidateAndUnpackCert` directly? I believe if no roots are provided, verification will fail so we don't need an explicit check, though it would be good to have a unit test to confirm.

Reply:
[FYI] I tried to "combine" the two functions but was getting test failures with cosign attach flows - likely I have missed some code paths.  Backed up to the state that you reviewed for now to get into the green build zone - will try again "with smaller steps."

Reply:
I got rid of the "special" function `ValidateAndUnpackCertWithCertPools` in [44e0ff5](https://github.com/sigstore/cosign/pull/3464/commits/44e0ff5f8a94844a5b94a9d31ba51689241735e1).
Do you think I should remove this check? 
https://github.com/sigstore/cosign/pull/3464/files#diff-fc9a26a4caa1d8684a4bdd9102f187f44e1f7fce7c06a19f4185035f13317cc8R296-R298
```go
                               if co.RootCerts == nil {
                                       return errors.New("no CA roots provided to validate certificate")
                               }
```
I would think it can be good to fail earlier with possibly more explicit error what have gone wrong, to make for easier troubleshooting (if it ever gets triggered).  Let me know if you prefer this yanked @haydentherapper.  I will also look into adding or expanding the unit tests, as you mentioned above.